powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OperationServiceBehavior.java

```diff
@@ -730,8 +731,8 @@ private boolean isDataEqual(@NotNull OperationEntity operation, String providedD
     }
 
     // Merge two maps into new one, replacing values in the first map when collision occurs
-    private Map<String, String> mapMerge(Map<String, String> m1, Map<String, String> m2) {
-        final Map<String, String> m3 = new HashMap<>();
+    private Map<String, Serializable> mapMerge(Map<String, Serializable> m1, Map<String, Serializable> m2) {
+        final Map<String, Serializable> m3 = new HashMap<>();
```

Comment:
Do we have a test for that? What will it look like? I mean, will the user-agent be a string or a structured map?